packages/desktop-client/src/hooks/useDisplayPayee.tsx

```diff
@@ -0,0 +1,188 @@
+import { createContext, type ReactNode, useContext, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { q } from 'loot-core/shared/query';
+import {
+  type AccountEntity,
+  type PayeeEntity,
+  type TransactionEntity,
+} from 'loot-core/types/models';
+
+import { useAccounts } from './useAccounts';
+import { usePayeesById } from './usePayees';
+import { useTransactions } from './useTransactions';
+
+type DisplayPayeeContextValue = {
+  displayPayees: Record<TransactionEntity['id'], string>;
+};
+
+const DisplayPayeeContext = createContext<DisplayPayeeContextValue | null>(
+  null,
+);
+
+type DisplayPayeeContextProviderProps = {
+  transactions: readonly TransactionEntity[];
+  children: ReactNode;
+};
+
+export function DisplayPayeeContextProvider({
```

Comment:
This queries all the subtransactions in one query and pre-calculates the display payees so the `useDisplayPayee` will only have to lookup the context and not run a query for each rendered transaction rows.